lib/home/widgets/collections_screen.dart

```diff
@@ -15,41 +15,88 @@ class CollectionsScreen extends StatefulWidget {
 class _CollectionsScreenState extends State<CollectionsScreen> {
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-        appBar: AppBar(title: const Text('Collections')),
-        body: Center(
-            child: Column(
-          children: [
-            // TODO: Implement dynamic Listview with multiple collections
-            const Divider(),
-            BlocBuilder<CollectionBloc, CollectionState>(
-              builder: (context, state) {
-                return ListTile(
-                    leading: const Icon(Icons.description),
-                    title: Text(state.gibsonsForm.respondentName.value),
-                    subtitle: Text(state.gibsonsForm.interviewDate.value),
-                    // TODO: Change to state.fourthPassStatus.isSubmissionSuccess
-                    trailing: state
-                            .gibsonsForm.sensitizationStatus.isSubmissionSuccess
-                        ? const Icon(Icons.done)
-                        : const Icon(Icons.pause),
-                    // TODO: Implement switching to last opened page
-                    onTap: () =>
-                        Navigator.pushNamed(context, PageRouter.collection));
-              },
-            ),
-            const Divider()
-          ],
-        )),
-        floatingActionButton: Column(
-            mainAxisAlignment: MainAxisAlignment.end,
-            crossAxisAlignment: CrossAxisAlignment.end,
-            children: <Widget>[
-              FloatingActionButton(
-                  child: const Icon(Icons.add),
-                  onPressed: () =>
-                      // TODO: Fix this such that a new collection will begin
-                      Navigator.pushNamed(context, PageRouter.collection))
-            ]));
+    return BlocBuilder<HomeBloc, HomeState>(
+      builder: (context, state) {
+        return Scaffold(
+            appBar: AppBar(title: const Text('Collections')),
+            body: ListView.builder(
+                padding: const EdgeInsets.all(2.0),
+                itemCount: state.gibsonsForms.length,
+                itemBuilder: (context, index) {
+                  // TODO: Refactor to a standalone widget
+                  return Card(
+                      // Unique key is needed to correctly rebuild after
+                      // deleting or modifying this widget and its subwidgets
+                      key: Key(state.gibsonsForms[index]!.id),
```

Comment:
There seems to be a key related bug when you go back to the collections page having not filled in the name or the date.
![collection_bug](https://user-images.githubusercontent.com/63778652/148174425-647dd108-21d2-497b-bbe3-f933ededfc36.jpg)

Reply:
Good catch! I'll go on it now.

Reply:
Yup, this also happens when the respondent Name is the same as the Collection Date (which hopefully shouldn't happen) so that needs to be fixed too.

I implemented it via a `UniqueKey()`, seems to have fixed it but honestly don't really know the best practices that much. From [here](https://youtu.be/kn0EOS-ZiIc) it correctly seems that keys must be specified for Lists of stateful widgets (as the `TextFormField`) so that they update properly (which seems to be the case here since if no keys are specified, modyfing the data doesn't rebuild the widget).